File: 6. ZigZag Conversion/ans.py
s = "PAYPALISHIRING"
numRows = 3

def convert(s, numRows):
    if numRows == 1: return s
    rows = ["" for _ in range(min(numRows,len(s)))]
    curRow = 0
    goingDown = False
    for c in s:
        rows[curRow] += c
        if curRow == 0 or curRow == numRows - 1:
            goingDown = not goingDown
        curRow += 1 if goingDown else -1
    ret = ""
    for row in rows:
        ret += row
    return ret
print(convert(s, numRows))

File: 6. ZigZag Conversion/ans.js
var convert = function(s, numRows) {
    if (numRows == 1) return s;
    
    var container = [];
    for (var i = 0; i < numRows; i++) container.push("");
    
    var up = true;
    var index = 0;
    
    for (str of s) {
        if (index == 0 || index == numRows-1) up = !up;
        
        if (up == false) {
            container[index] += str;
            index++;
        } else {         
            container[index] += str;
            index--;
        }
    }
    
    return container.join("");
};

const [s, numRows] = ["PAYPALISHIRING", 3];

console.log(convert(s, numRows)); //"PAHNAPLSIIGYIR"